kore/app/share/GlobalMain.hs

```diff
@@ -75,6 +83,8 @@ import qualified Kore.Verified as Verified
 import qualified Paths_kore as MetaData
                  ( version )
 
+type GlobalMainIO = ReaderT (LogAction IO Logger.LogMessage) IO
```

Comment:
I would prefer to use a `newtype` here, so that we don't accidentally depend on the implementation details. You might have to `derive` several instances; at least `MonadReader` and `MonadIO` (maybe others), but if it gets much harder than deriving instances then you can forget I said anything.

We might also simply name this `Main` instead, unless @vladciobanu feels differently.